contracts/internal/EverlongERC4626.sol

```diff
@@ -54,6 +34,97 @@ abstract contract EverlongERC4626 is ERC4626 {
         _decimals = success ? result : _DEFAULT_UNDERLYING_DECIMALS;
     }
 
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Stateful                                                │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    /// @inheritdoc ERC4626
+    function deposit(
+        uint256 assets,
+        address to
+    ) public override returns (uint256 shares) {
+        IERC20(_asset).approve(hyperdrive, assets);
+        shares = super.deposit(assets, to);
+    }
+
+    /// @inheritdoc ERC4626
+    function redeem(
+        uint256 shares,
+        address to,
+        address owner
+    ) public override returns (uint256 assets) {
+        assets = super.redeem(shares, to, owner);
+        rebalance();
```

Comment:
Wouldn't it make more sense to close matured positions first? Unless I'm missing something, this is just going to pay out some idle to the user, which the pool won't have unless it closes existing positions. I understand why we'd want to put the remaining amount of capital into new positions afterwards, so we'd probably still keep this call afterwards.

Something else that I'd say here is that I get that the logic is simple, but each of these lines of code is pretty consequential and the order matters. IMO, both of these lines should be commented with a high-level description of what's happening and why it needs to happen here.

Reply:
I think that closing matured positions prior to redemption is handled in the [`_beforeWithdraw()`](https://github.com/delvtech/everlong/blob/4bac426c161cb0393b17fca902dcf064895c137e/contracts/internal/EverlongERC4626.sol#L98-L101) hook... but could be mistaken.

Will add high-level description + comments